types/original/index.d.ts

```diff
@@ -0,0 +1,27 @@
+// Type definitions for original 1.0
+// Project: https://github.com/unshiftio/original
+// Definitions by: Wayne Carson <https://github.com/wcarson>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * Transform an URL to a valid origin value.
+ *
+ * @param url URL to transform to it's origin.
+ * @returns The origin.
+ * @api public
+ */
+declare function origin(url: string|Object): string;
```

Comment:
I wouldn't use Object here &mdash; and removing it allows you to remove `"ban-types": false` from tslint.json.

From looking at the source, it looks like a better type would be `string | object | { protocol: string, host: string }`

Reply:
Updated param type per your suggestion. Thanks!